lib/pmdtester/builders/pmd_report_builder.rb

```diff
@@ -27,21 +27,29 @@ def execute_reset_cmd(type, tag)
         reset_cmd = "git reset --hard #{tag}"
       when 'hg'
         reset_cmd = "hg up #{tag}"
-      else
-        raise Exception, "Unknown #{type} repository"
       end
 
       Cmd.execute(reset_cmd)
     end
 
     def get_projects
-      puts 'Cloning projects started'
+      logger.info 'Cloning projects started'
 
       @projects.each do |project|
         path = project.local_source_path
         clone_cmd = "#{project.type} clone #{project.connection} #{path}"
         if File.exist?(path)
-          puts "Skipping clone, project path #{path} already exists"
+          logger.warn "Skipping clone, project path #{path} already exists"
+          update_cmd = "#{project.type} pull #{project.connection}"
```

Comment:
Good, you added the update :+1: 

Does this already work? For git, I think, we need to specify a branch, from which we pull? (since project.connection is a URL).
As we cloned it earlier, we could rely on the defaults and just do a "git pull origin".... which hopefully updates the master branch...
For mercurial, I'm not sure...

Reply:
I thought about it carefully, we shouldn't update the warehouse, we should choose the constant branch of the open source project.This will ensure that baselines are generated based on the same source code especially when it comes to online mode.